pkg/machines/components/networks/createNetworkDialog.jsx

```diff
@@ -0,0 +1,560 @@
+    const validationEnd = validationFailed['ipv' + ipVersion + 'DhcpRangeEnd'] ? 'error' : undefined;
+
+    return (
+        <React.Fragment>
+            <label className='checkbox-inline'>
+                <input id={'network-ipv' + ipVersion + '-dhcp'}
+                    type='checkbox'
+                    checked={expanded}
+                    onChange={e => onValueChanged('ipv' + ipVersion + 'DhcpEnabled', !expanded)} />
+                {_("Set DHCP Range")}
+            </label>
+
+            {expanded && <React.Fragment>
+                <div className='create-network-dialog-grid'>
+                    <div className='ct-form'>
+                        <label className='control-label'> {_("Start")} </label>
+                        <FormGroup validationState={validationStart} controlId={'ipv' + ipVersion + '-dhcp-range-start'}>
+                            <input
+                               id={'network-ipv' + ipVersion + '-dhcp-range-start'}
+                               type='text'
+                               value={rangeStart}
+                               onChange={e => onValueChanged('ipv' + ipVersion + 'DhcpRangeStart', e.target.value)}
+                               className='form-control' />
+                            { validationStart == 'error' &&
+                            <HelpBlock>
+                                <p className='text-danger'>{validationFailed['ipv' + ipVersion + 'DhcpRangeStart']}</p>
+                            </HelpBlock> }
+                        </FormGroup>
+                    </div>
+                    <div className='ct-form'>
+                        <label className='control-label'> {_("End")} </label>
+                        <FormGroup validationState={validationEnd} controlId={'ipv' + ipVersion + '-dhcp-range-end'}>
+                            <input
+                               id={'network-ipv' + ipVersion + '-dhcp-range-end'}
+                               type='text'
+                               value={rangeEnd}
+                               onChange={e => onValueChanged('ipv' + ipVersion + 'DhcpRangeEnd', e.target.value)}
+                               className='form-control' />
+                            { validationEnd == 'error' &&
+                            <HelpBlock>
+                                <p className='text-danger'>{validationFailed['ipv' + ipVersion + 'DhcpRangeEnd']}</p>
+                            </HelpBlock> }
+                        </FormGroup>
+                    </div>
+                </div>
+            </React.Fragment> }
+        </React.Fragment>
+    );
+};
+
+const Ipv4Row = ({ validationFailed, dialogValues, onValueChanged }) => {
+    const validationAddress = validationFailed.ipv4 ? 'error' : undefined;
+    const validationNetmask = validationFailed.netmask ? 'error' : undefined;
+
+    return (
+        <React.Fragment>
+            <div className='create-network-dialog-grid'>
+                <div>
```

Comment:
Is this div really needed. Maybe you can replace it with <React.Fragment> ?
@garrett really hates extra unneeded divs :)

Reply:
Oh this is just temporary until ct-form supports labels on top. It was mentioned in @garrett  comment when talking about design with andreas.